### PR TITLE
Sort profile bundle/charm lists by name

### PR DESCRIPTION
**Pull request by huwshimi**
Fixes: #3507.

### CODE REVIEW

**Comment by huwshimi** [comment]
QA:
- open a profile with a bunch of charms/bundles
- check that the charms/bundles are sorted by name

**Comment by huwshimi** [comment]
[Sort charms/bundles by name by default](https://trello.com/c/AgAcPIxP/367-sort-charms-bundles-by-name-by-default)

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/juju-gui/8549/

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/juju-gui/8550/

File: jujugui/static/gui/src/app/components/profile/bundle-list/bundle-list.js [hunk 1]
```diff
@@ -101,6 +101,22 @@ class ProfileBundleList extends React.Component {
       </h2>);
   }
 
+  /**
+    Sort by the key attribute.
+    @param {Object} a The first value.
+    @param {Object} b The second value.
+    @returns {Array} The sorted array.
+  */
+  _byName(a, b) {
```
Comment (hatched) [inline jujugui/static/gui/src/app/components/profile/bundle-list/bundle-list.js]: Can this be moved into the BasicTable? The sort key could then take a string or a function?

File: jujugui/static/gui/src/app/components/profile/bundle-list/bundle-list.js [hunk 2]
```diff
@@ -200,6 +216,7 @@ class ProfileBundleList extends React.Component {
                     {version}
                   </div>
                 </div>)} />),
+          extraData: bundle.name,
```
Comment (hatched) [inline jujugui/static/gui/src/app/components/profile/bundle-list/bundle-list.js]: what about 'sortData'?

File: jujugui/static/gui/src/app/components/profile/bundle-list/bundle-list.js (hunk 1 above)
Comment (huwshimi) [reply jujugui/static/gui/src/app/components/profile/bundle-list/bundle-list.js]: This is the existing way that the table handles sorting.

File: jujugui/static/gui/src/app/components/profile/bundle-list/bundle-list.js (hunk 2 above)
Comment (huwshimi) [reply jujugui/static/gui/src/app/components/profile/bundle-list/bundle-list.js]: This is the key the table finds the extra sort info from.

**Comment by hatched** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**